fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Modifier.kt

```diff
@@ -49,3 +56,30 @@ fun Modifier.dashedBorder(
         )
     },
 )
+
+/**
+ * Used when clickable needs to be debounced to prevent rapid successive clicks
+ * from calling the onClick function.
+ *
+ * @param debounceInterval The length of time to wait between click events in milliseconds
+ * @param onClick Callback for when item this modifier effects is clicked
+ */
+fun Modifier.debouncedClickable(
+    debounceInterval: Long = 1000L,
+    onClick: () -> Unit,
+) = composed(
+    factory = {
```

Comment:
nit: Since `composed` is a [SAM](https://kotlinlang.org/docs/fun-interfaces.html#sam-conversions) if we remove the `factory` named parameter here we can also remove the parens for the function call. Looks a little cleaner IMO but that's not a super strong opinion

Reply:
ah yes, good call, removed the `factory` parameter